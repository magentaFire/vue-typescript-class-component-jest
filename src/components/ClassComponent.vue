<template>
    <div class="class-component">{{ message }}{{ myProp }}</div>
</template>

<script lang="ts">
import Vue from "vue";
import Component from "vue-class-component";

@Component({
    created() {
        console.log("created")
    },
    props: ["myProp"]
})
class ClassComponent extends Vue {

    message: string = "";

}

export default ClassComponent;
</script>