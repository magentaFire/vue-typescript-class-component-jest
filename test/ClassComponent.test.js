import { mount } from "vue-test-utils";
import Vue from "vue";
import ClassComponent from "../src/components/ClassComponent.vue";

describe("ClassComponent.test.ts", () => {
    let wrapper;

    beforeEach(() => {
        wrapper = mount(ClassComponent, {
            data: {
                message: "Hello world",
            },
            propsData: {
                myProp: "!"
            }
        })
    })

    it(".message = 'Hello world'", () => {
        expect(wrapper.vm.message).toEqual("Hello world");
    })

    it(".myProp = '!'", () => {
        expect(wrapper.vm.myProp).toEqual("!");
    })

    it("updates the .textContent to 'foo'", done => {
        wrapper.vm.message = "foo"

        // vuejs updates the dom asynchronously
        // wait for the nextTick callback before running the test
        Vue.nextTick(() => {
            try {
                expect(wrapper.vm.$el.textContent).toBe('foo!')
                done()
            } catch (err) {
                done.fail(err)
            }
        })
    })

    it("has the expected html structure", () => {
        expect(wrapper.vm.$el).toMatchSnapshot()
    })
})