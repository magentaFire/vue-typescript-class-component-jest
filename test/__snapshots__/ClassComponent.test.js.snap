// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassComponent.test.ts has the expected html structure 1`] = `
<div
  class="class-component"
>
  Hello world!
</div>
`;
